<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='read_report() imports data from an MGI report into R as a tidy data set.
You may call this function in two alternative ways:
Using report_key: this is the easiest approach. A report key maps to a
report currently hosted at MGI, e.g. read_report("marker_list2") reads
MRK_List2.rpt directly from MGI server into R. See Supported Reports below
for options.
Using report_file and report_type: this approach is more flexible as
you can read directly from a file or URL.


Supported Reports


The set of currently supported reports:
reports
#&amp;gt; # A tibble: 13 x 4
#&amp;gt;    report_key                 report_file              report_type   report_name
#&amp;gt;    &amp;lt;chr&amp;gt;                      &amp;lt;chr&amp;gt;                    &amp;lt;chr&amp;gt;         &amp;lt;chr&amp;gt;      
#&amp;gt;  1 marker_list1               MRK_List1.rpt            MRK_List1     Mouse Gene~
#&amp;gt;  2 marker_list2               MRK_List2.rpt            MRK_List2     Mouse Gene~
#&amp;gt;  3 marker_coordinates         MGI_MRK_Coord.rpt        MGI_MRK_Coord MGI Marker~
#&amp;gt;  4 gene_model_coordinates     MGI_Gene_Model_Coord.rpt MGI_Gene_Mod~ MGI Gene M~
#&amp;gt;  5 sequence_coordinates       MGI_GTGUP.gff            MGI_GTGUP     MGI Sequen~
#&amp;gt;  6 genbank_refseq_ensembl_ids MRK_Sequence.rpt         MRK_Sequence  MGI Marker~
#&amp;gt;  7 swiss_trembl_ids           MRK_SwissProt_TrEMBL.rpt MRK_SwissPro~ MGI Marker~
#&amp;gt;  8 swiss_prot_ids             MRK_SwissProt.rpt        MRK_SwissProt MGI Marker~
#&amp;gt;  9 gene_trap_ids              MRK_GeneTrap.rpt         MRK_GeneTrap  MGI Marker~
#&amp;gt; 10 ensembl_ids                MRK_ENSEMBL.rpt          MRK_ENSEMBL   MGI Marker~
#&amp;gt; 11 biotype_conflicts          MGI_BioTypeConflict.rpt  MGI_BioTypeC~ MGI Marker~
#&amp;gt; 12 primers                    PRB_PrimerSeq.rpt        PRB_PrimerSeq MGI Marker~
#&amp;gt; 13 interpro_domains           MGI_InterProDomains.rpt  MGI_InterPro~ InterPro d~


'><title>Read an MGI report — read_report • mgi.report.reader</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.9/font.css" rel="stylesheet"><link href="../deps/Inconsolata-0.4.9/font.css" rel="stylesheet"><link href="../deps/Source_Sans_Pro-0.4.9/font.css" rel="stylesheet"><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Read an MGI report — read_report"><meta property="og:description" content='read_report() imports data from an MGI report into R as a tidy data set.
You may call this function in two alternative ways:
Using report_key: this is the easiest approach. A report key maps to a
report currently hosted at MGI, e.g. read_report("marker_list2") reads
MRK_List2.rpt directly from MGI server into R. See Supported Reports below
for options.
Using report_file and report_type: this approach is more flexible as
you can read directly from a file or URL.


Supported Reports


The set of currently supported reports:
reports
#&amp;gt; # A tibble: 13 x 4
#&amp;gt;    report_key                 report_file              report_type   report_name
#&amp;gt;    &amp;lt;chr&amp;gt;                      &amp;lt;chr&amp;gt;                    &amp;lt;chr&amp;gt;         &amp;lt;chr&amp;gt;      
#&amp;gt;  1 marker_list1               MRK_List1.rpt            MRK_List1     Mouse Gene~
#&amp;gt;  2 marker_list2               MRK_List2.rpt            MRK_List2     Mouse Gene~
#&amp;gt;  3 marker_coordinates         MGI_MRK_Coord.rpt        MGI_MRK_Coord MGI Marker~
#&amp;gt;  4 gene_model_coordinates     MGI_Gene_Model_Coord.rpt MGI_Gene_Mod~ MGI Gene M~
#&amp;gt;  5 sequence_coordinates       MGI_GTGUP.gff            MGI_GTGUP     MGI Sequen~
#&amp;gt;  6 genbank_refseq_ensembl_ids MRK_Sequence.rpt         MRK_Sequence  MGI Marker~
#&amp;gt;  7 swiss_trembl_ids           MRK_SwissProt_TrEMBL.rpt MRK_SwissPro~ MGI Marker~
#&amp;gt;  8 swiss_prot_ids             MRK_SwissProt.rpt        MRK_SwissProt MGI Marker~
#&amp;gt;  9 gene_trap_ids              MRK_GeneTrap.rpt         MRK_GeneTrap  MGI Marker~
#&amp;gt; 10 ensembl_ids                MRK_ENSEMBL.rpt          MRK_ENSEMBL   MGI Marker~
#&amp;gt; 11 biotype_conflicts          MGI_BioTypeConflict.rpt  MGI_BioTypeC~ MGI Marker~
#&amp;gt; 12 primers                    PRB_PrimerSeq.rpt        PRB_PrimerSeq MGI Marker~
#&amp;gt; 13 interpro_domains           MGI_InterProDomains.rpt  MGI_InterPro~ InterPro d~


'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-none" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mgi.report.reader</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-concepts">Concepts</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-concepts">
    <a class="dropdown-item" href="../articles/genetic_marker.html">Genetic Marker</a>
    <a class="dropdown-item" href="../articles/representative_sequence.html">Representative Genomic Sequence</a>
  </div>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-reports">Reports</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-reports">
    <h6 class="dropdown-header" data-toc-skip>Mouse Genetic Markers</h6>
    <a class="dropdown-item" href="../articles/marker_list1.html">All Markers (MRK_List1.rpt)</a>
    <a class="dropdown-item" href="../articles/marker_list2.html">Official Markers (MRK_List2.rpt)</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Sequence Data</h6>
    <a class="dropdown-item" href="../articles/marker_coordinates.html">Marker Coordinates (MGI_MRK_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_model_coordinates.html">Gene Model Coordinates (MGI_Gene_Model_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/sequence_coordinates.html">Sequence Coordinates (MGI_GTGUP.gff)</a>
    <a class="dropdown-item" href="../articles/genbank_refseq_ensembl_ids.html">GenBank, RefSeq and Ensembl IDs (MRK_Sequence.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_trembl_ids.html">UniProtKB Swiss-Prot and TrEMBL (MRK_SwissProt_TrEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_prot_ids.html">UniProtKB Swiss-Prot (MRK_SwissProt.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_trap_ids.html">Gene Traps (MRK_GeneTrap.rpt)</a>
    <a class="dropdown-item" href="../articles/ensembl_ids.html">Ensembl IDs (MRK_ENSEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/biotype_conflicts.html">Ensembl/NCBI Biotype Conflicts (MGI_BioTypeConflict.rpt)</a>
    <a class="dropdown-item" href="../articles/primers.html">Primer Pairs (PRB_PrimerSeq.rpt)</a>
    <a class="dropdown-item" href="../articles/interpro_domains.html">InterPro Domains (MGI_InterProDomains.rpt)</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/patterninstitute/mgi.report.reader/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Read an MGI report</h1>
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/mgi.report.reader/blob/HEAD/R/read_report.R" class="external-link"><code>R/read_report.R</code></a></small>
      <div class="d-none name"><code>read_report.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>read_report()</code> imports data from an MGI report into R as a tidy data set.</p>
<p>You may call this function in two alternative ways:</p><ul><li><p>Using <code>report_key</code>: this is the easiest approach. A report key maps to a
report currently hosted at MGI, e.g. <code>read_report("marker_list2")</code> reads
<code>MRK_List2.rpt</code> directly from MGI server into R. See Supported Reports below
for options.</p></li>
<li><p>Using <code>report_file</code> and <code>report_type</code>: this approach is more flexible as
you can read directly from a file or URL.</p></li>
</ul><div class="section">
<h3 id="supported-reports">Supported Reports<a class="anchor" aria-label="anchor" href="#supported-reports"></a></h3>


<p>The set of currently supported reports:</p>
<p></p><div class="sourceCode r"><pre><code><span><span class="va">reports</span></span>
<span><span class="co">#&gt; # A tibble: 13 x 4</span></span>
<span><span class="co">#&gt;    report_key                 report_file              report_type   report_name</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                      &lt;chr&gt;                    &lt;chr&gt;         &lt;chr&gt;      </span></span>
<span><span class="co">#&gt;  1 marker_list1               MRK_List1.rpt            MRK_List1     Mouse Gene~</span></span>
<span><span class="co">#&gt;  2 marker_list2               MRK_List2.rpt            MRK_List2     Mouse Gene~</span></span>
<span><span class="co">#&gt;  3 marker_coordinates         MGI_MRK_Coord.rpt        MGI_MRK_Coord MGI Marker~</span></span>
<span><span class="co">#&gt;  4 gene_model_coordinates     MGI_Gene_Model_Coord.rpt MGI_Gene_Mod~ MGI Gene M~</span></span>
<span><span class="co">#&gt;  5 sequence_coordinates       MGI_GTGUP.gff            MGI_GTGUP     MGI Sequen~</span></span>
<span><span class="co">#&gt;  6 genbank_refseq_ensembl_ids MRK_Sequence.rpt         MRK_Sequence  MGI Marker~</span></span>
<span><span class="co">#&gt;  7 swiss_trembl_ids           MRK_SwissProt_TrEMBL.rpt MRK_SwissPro~ MGI Marker~</span></span>
<span><span class="co">#&gt;  8 swiss_prot_ids             MRK_SwissProt.rpt        MRK_SwissProt MGI Marker~</span></span>
<span><span class="co">#&gt;  9 gene_trap_ids              MRK_GeneTrap.rpt         MRK_GeneTrap  MGI Marker~</span></span>
<span><span class="co">#&gt; 10 ensembl_ids                MRK_ENSEMBL.rpt          MRK_ENSEMBL   MGI Marker~</span></span>
<span><span class="co">#&gt; 11 biotype_conflicts          MGI_BioTypeConflict.rpt  MGI_BioTypeC~ MGI Marker~</span></span>
<span><span class="co">#&gt; 12 primers                    PRB_PrimerSeq.rpt        PRB_PrimerSeq MGI Marker~</span></span>
<span><span class="co">#&gt; 13 interpro_domains           MGI_InterProDomains.rpt  MGI_InterPro~ InterPro d~</span></span></code></pre><p></p></div>
</div>

    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">read_report</span><span class="op">(</span></span>
<span>  report_key <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  report_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  report_type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  n_max <span class="op">=</span> <span class="cn">Inf</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>report_key</dt>
<dd><p>A character vector. A key used to uniquely refer to an MGI
report.</p></dd>


<dt>report_file</dt>
<dd><p>A character vector. The file path or URL to an MGI report
file.</p></dd>


<dt>report_type</dt>
<dd><p>A character vector. The type of an MGI report.</p></dd>


<dt>n_max</dt>
<dd><p>Maximum number of lines to read.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <a href="https://tibble.tidyverse.org/reference/tibble-package.html" class="external-link">tibble</a> with report data in tidy
format. The set of variables is dependent on the specific report requested:</p><ul><li><p>For <code>"marker_list1"</code>, see <code><a href="../articles/marker_list1.html">vignette("marker_list1")</a></code>.</p></li>
<li><p>For <code>"marker_list2"</code>, see <code><a href="../articles/marker_list2.html">vignette("marker_list2")</a></code>.</p></li>
<li><p>For <code>"marker_coordinates"</code>, see <code><a href="../articles/marker_coordinates.html">vignette("marker_coordinates")</a></code>.</p></li>
<li><p>For <code>"gene_model_coordinates"</code>, see <code><a href="../articles/gene_model_coordinates.html">vignette("gene_model_coordinates")</a></code>.</p></li>
<li><p>For <code>"sequence_coordinates"</code>, see <code><a href="../articles/sequence_coordinates.html">vignette("sequence_coordinates")</a></code>.</p></li>
<li><p>For <code>"genbank_refseq_ensembl_ids"</code>, see <code><a href="../articles/genbank_refseq_ensembl_ids.html">vignette("genbank_refseq_ensembl_ids")</a></code>.</p></li>
<li><p>For <code>"swiss_trembl_ids"</code>, see <code><a href="../articles/swiss_trembl_ids.html">vignette("swiss_trembl_ids")</a></code>.</p></li>
<li><p>For <code>"swiss_prot_ids"</code>, see <code><a href="../articles/swiss_prot_ids.html">vignette("swiss_prot_ids")</a></code>.</p></li>
<li><p>For <code>"gene_trap_ids"</code>, see <code><a href="../articles/gene_trap_ids.html">vignette("gene_trap_ids")</a></code>.</p></li>
<li><p>For <code>"ensembl_ids"</code>, see <code><a href="../articles/ensembl_ids.html">vignette("ensembl_ids")</a></code>.</p></li>
<li><p>For <code>"biotype_conflicts"</code>, see <code><a href="../articles/biotype_conflicts.html">vignette("biotype_conflicts")</a></code>.</p></li>
<li><p>For <code>"primers"</code>, see <code><a href="../articles/primers.html">vignette("primers")</a></code>.</p></li>
<li><p>For <code>"interpro_domains"</code>, see <code><a href="../articles/interpro_domains.html">vignette("interpro_domains")</a></code>.</p></li>
</ul></div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ramiro Magno, David Shaw, Isabel Duarte, Ismail Gbadamosi, Ali Jawaid, Nencki Institute of Experimental Biology, University of Algarve, The Jackson Laboratory, Pattern Institute.</p>
</div>

<div class="pkgdown-footer-right">
  
</div>

    </footer></div>

  

  

  </body></html>

