<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="reports is a data set of supported MGI reports, meaning reports that
{mgi.report.reader} can currently read into R.
To browse all reports made available by MGI visit
https://www.informatics.jax.org/downloads/reports."><title>Supported MGI reports — reports • mgi.report.reader</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.9/font.css" rel="stylesheet"><link href="../deps/Inconsolata-0.4.9/font.css" rel="stylesheet"><link href="../deps/Source_Sans_Pro-0.4.9/font.css" rel="stylesheet"><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Supported MGI reports — reports"><meta property="og:description" content="reports is a data set of supported MGI reports, meaning reports that
{mgi.report.reader} can currently read into R.
To browse all reports made available by MGI visit
https://www.informatics.jax.org/downloads/reports."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-none" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mgi.report.reader</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-concepts">Concepts</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-concepts">
    <a class="dropdown-item" href="../articles/genetic_marker.html">Genetic Marker</a>
    <a class="dropdown-item" href="../articles/representative_sequence.html">Representative Genomic Sequence</a>
  </div>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-reports">Reports</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-reports">
    <h6 class="dropdown-header" data-toc-skip>Mouse Genetic Markers</h6>
    <a class="dropdown-item" href="../articles/marker_list1.html">All Markers (MRK_List1.rpt)</a>
    <a class="dropdown-item" href="../articles/marker_list2.html">Official Markers (MRK_List2.rpt)</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Sequence Data</h6>
    <a class="dropdown-item" href="../articles/marker_coordinates.html">Marker Coordinates (MGI_MRK_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_model_coordinates.html">Gene Model Coordinates (MGI_Gene_Model_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/sequence_coordinates.html">Sequence Coordinates (MGI_GTGUP.gff)</a>
    <a class="dropdown-item" href="../articles/genbank_refseq_ensembl_ids.html">GenBank, RefSeq and Ensembl IDs (MRK_Sequence.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_trembl_ids.html">UniProtKB Swiss-Prot and TrEMBL (MRK_SwissProt_TrEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_prot_ids.html">UniProtKB Swiss-Prot (MRK_SwissProt.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_trap_ids.html">Gene Traps (MRK_GeneTrap.rpt)</a>
    <a class="dropdown-item" href="../articles/ensembl_ids.html">Ensembl IDs (MRK_ENSEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/biotype_conflicts.html">Ensembl/NCBI Biotype Conflicts (MGI_BioTypeConflict.rpt)</a>
    <a class="dropdown-item" href="../articles/primers.html">Primer Pairs (PRB_PrimerSeq.rpt)</a>
    <a class="dropdown-item" href="../articles/interpro_domains.html">InterPro Domains (MGI_InterProDomains.rpt)</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/patterninstitute/mgi.report.reader/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Supported MGI reports</h1>
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/mgi.report.reader/blob/HEAD/R/reports.R" class="external-link"><code>R/reports.R</code></a></small>
      <div class="d-none name"><code>reports.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>reports</code> is a data set of supported MGI reports, meaning reports that
<code>{mgi.report.reader}</code> can currently read into R.
To browse all reports made available by MGI visit
<a href="https://www.informatics.jax.org/downloads/reports" class="external-link">https://www.informatics.jax.org/downloads/reports</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">reports</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A <a href="https://tibble.tidyverse.org/reference/tibble-package.html" class="external-link">tibble</a> of 4 variables:</p>
<dl><dt><code>report_key</code></dt>
<dd><p>A string key used to uniquely refer to an MGI report, which is only meaningful within the context of the <code>{mgi.report.reader}</code>.</p></dd>

<dt><code>report_file</code></dt>
<dd><p>MGI report file name as hosted at <a href="https://www.informatics.jax.org/downloads/reports%20" class="external-link"> https://www.informatics.jax.org/downloads/reports </a>.</p></dd>

<dt><code>report_type</code></dt>
<dd><p>MGI report type. The type is used internally to find the appropriate reader for parsing, and is only meaningful within the context of <code>{mgi.report.reader}</code>.</p></dd>

<dt><code>report_name</code></dt>
<dd><p>MGI report name. Report names are taken from <a href="https://www.informatics.jax.org/downloads/reports/index.html%20" class="external-link"> https://www.informatics.jax.org/downloads/reports/index.html </a>.</p></dd>


</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">reports</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 13 × 4</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    report_key                 report_file              report_type   report_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> marker_list1               MRK_List1.rpt            MRK_List1     Mouse Gene…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> marker_list2               MRK_List2.rpt            MRK_List2     Mouse Gene…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> marker_coordinates         MGI_MRK_Coord.rpt        MGI_MRK_Coord MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> gene_model_coordinates     MGI_Gene_Model_Coord.rpt MGI_Gene_Mod… MGI Gene M…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> sequence_coordinates       MGI_GTGUP.gff            MGI_GTGUP     MGI Sequen…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> genbank_refseq_ensembl_ids MRK_Sequence.rpt         MRK_Sequence  MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> swiss_trembl_ids           MRK_SwissProt_TrEMBL.rpt MRK_SwissPro… MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> swiss_prot_ids             MRK_SwissProt.rpt        MRK_SwissProt MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> gene_trap_ids              MRK_GeneTrap.rpt         MRK_GeneTrap  MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> ensembl_ids                MRK_ENSEMBL.rpt          MRK_ENSEMBL   MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span> biotype_conflicts          MGI_BioTypeConflict.rpt  MGI_BioTypeC… MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span> primers                    PRB_PrimerSeq.rpt        PRB_PrimerSeq MGI Marker…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span> interpro_domains           MGI_InterProDomains.rpt  MGI_InterPro… InterPro d…</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ramiro Magno, David Shaw, Isabel Duarte, Ismail Gbadamosi, Ali Jawaid, Nencki Institute of Experimental Biology, University of Algarve, The Jackson Laboratory, Pattern Institute.</p>
</div>

<div class="pkgdown-footer-right">
  
</div>

    </footer></div>

  

  

  </body></html>

