<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mgi.report.reader">
<title>GenBank, RefSeq and Ensembl IDs • mgi.report.reader</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Inconsolata-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Source_Sans_Pro-0.4.9/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GenBank, RefSeq and Ensembl IDs">
<meta property="og:description" content="mgi.report.reader">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-none" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mgi.report.reader</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.9.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-reports">Reports</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-reports">
    <h6 class="dropdown-header" data-toc-skip>Mouse Genetic Markers</h6>
    <a class="dropdown-item" href="../articles/marker_list1.html">All Markers (MRK_List1.rpt)</a>
    <a class="dropdown-item" href="../articles/marker_list2.html">Official Markers (MRK_List2.rpt)</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Sequence Data</h6>
    <a class="dropdown-item" href="../articles/marker_coordinates.html">Marker Coordinates (MGI_MRK_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_model_coordinates.html">Gene Model Coordinates (MGI_Gene_Model_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/sequence_coordinates.html">Sequence Coordinates (MGI_GTGUP.gff)</a>
    <a class="dropdown-item" href="../articles/genbank_refseq_ensembl_ids.html">GenBank, RefSeq and Ensembl IDs (MRK_Sequence.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_trembl_ids.html">UniProtKB Swiss-Prot and TrEMBL (MRK_SwissProt_TrEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_prot_ids.html">UniProtKB Swiss-Prot (MRK_SwissProt.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_trap_ids.html">Gene Traps (MRK_GeneTrap.rpt)</a>
    <a class="dropdown-item" href="../articles/ensembl_ids.html">Ensembl IDs (MRK_ENSEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/biotype_conflicts.html">Ensembl/NCBI Biotype Conflicts (MGI_BioTypeConflict.rpt)</a>
    <a class="dropdown-item" href="../articles/primers.html">Primer Pairs (PRB_PrimerSeq.rpt)</a>
    <a class="dropdown-item" href="../articles/interpro_domains.html">InterPro Domains (MGI_InterProDomains.rpt)</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/patterninstitute/mgi.report.reader/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>GenBank, RefSeq and Ensembl IDs</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/mgi.report.reader/blob/HEAD/vignettes/articles/genbank_refseq_ensembl_ids.Rmd" class="external-link"><code>vignettes/articles/genbank_refseq_ensembl_ids.Rmd</code></a></small>
      <div class="d-none name"><code>genbank_refseq_ensembl_ids.Rmd</code></div>
    </div>

    
    
<p>The MGI report <code>MRK_Sequence.rpt</code> provides associations
between MGI genetic markers and GenBank, RefSeq, Ensembl and UniProtKB
identifiers.</p>
<p>To read this report using the key
<code>"genbank_refseq_ensembl_ids"</code>, use the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To read all records (more than 70,000), use `read_report("genbank_refseq_ensembl_ids")`.</span></span>
<span><span class="op">(</span><span class="va">assoc_to_seq_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_report.html">read_report</a></span><span class="op">(</span>report_key <span class="op">=</span> <span class="st">"genbank_refseq_ensembl_ids"</span>, n_max <span class="op">=</span> <span class="fl">30L</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 30 × 19</span></span></span>
<span><span class="co">##    marker_status marker_type marker_id   marker_symbol marker_name  feature_type</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> O             BAC/YAC end MGI:1341858 03B03F        DNA segment… BAC/YAC end </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> O             BAC/YAC end MGI:1341869 03B03R        DNA segment… BAC/YAC end </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> O             Gene        MGI:1918911 0610005C13Rik RIKEN cDNA … lncRNA gene </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> O             Gene        MGI:1923503 0610006L08Rik RIKEN cDNA … lncRNA gene </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> O             Gene        MGI:1925547 0610008J02Rik RIKEN cDNA … lncRNA gene </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> O             Gene        MGI:3698435 0610009E02Rik RIKEN cDNA … lncRNA gene </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> O             Gene        MGI:1918921 0610009F21Rik RIKEN cDNA … lncRNA gene </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> O             Gene        MGI:1918931 0610009K14Rik RIKEN cDNA … lncRNA gene </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> O             Gene        MGI:1914088 0610009L18Rik RIKEN cDNA … lncRNA gene </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> O             Gene        MGI:1915609 0610010K14Rik RIKEN cDNA … protein cod…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 20 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 13 more variables: chromosome &lt;fct&gt;, start &lt;int&gt;, end &lt;int&gt;, strand &lt;fct&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   genetic_map_pos &lt;dbl&gt;, genbank_id &lt;list&gt;, refseq_trp_id &lt;list&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   refseq_prt_id &lt;list&gt;, ensembl_trp_id &lt;list&gt;, ensembl_prt_id &lt;list&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   swiss_prt_id &lt;list&gt;, tr_embl_prt_id &lt;list&gt;, unigene_id &lt;chr&gt;</span></span></span></code></pre>
<div class="section level2">
<h2 id="genbank-refseq-ensembl-and-uniprotkb-identifiers">GenBank, RefSeq, Ensembl and UniProtKB identifiers<a class="anchor" aria-label="anchor" href="#genbank-refseq-ensembl-and-uniprotkb-identifiers"></a>
</h2>
<p>These variables hold one or more identifiers associated with each
genetic marker:</p>
<ul>
<li>
<code>genbank_id</code>: GenBank identifier(s)</li>
<li>
<code>refseq_trp_id</code>: RefSeq transcript identifier(s)</li>
<li>
<code>refseq_prt_id</code>: RefSeq protein identifier(s)</li>
<li>
<code>ensembl_trp_id</code>: Ensembl transcript identifier(s)</li>
<li>
<code>ensembl_prt_id</code>: Ensembl protein identifier(s)</li>
<li>
<code>swiss_prt_id</code>: UniProtKB/Swiss-Prot identifier(s)</li>
<li>
<code>tr_embl_prt_id</code>: UniProtKB/TrEMBL identifier(s)</li>
<li>
<code>unigene_id</code>: UniGene identifier<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;All values are &lt;code&gt;NA&lt;/code&gt; for this variable.&lt;/p&gt;"><sup>1</sup></a>
</li>
</ul>
<p>Except for <code>unigene_id</code>, all these variables are <a href="https://r4ds.hadley.nz/rectangling#list-columns" class="external-link">list-columns</a>,
and provide, potentially, multiple values for the same genetic
marker.</p>
<p>Having these data nested in list-columns offers the convenience of
having a table whose records (rows) correspond to one genetic marker.
The downside is that these multiple identifiers are not readily
accessible as they would be if stored in atomic columns. To unnest them
you may use <code><a href="https://tidyr.tidyverse.org/reference/unnest_longer.html" class="external-link">tidyr::unnest_longer()</a></code><a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;To learn more about how to rectangle your hierarchical
(nested) data, see &lt;a href="https://tidyr.tidyverse.org/articles/rectangle.html" class="external-link"&gt;Rectangling&lt;/a&gt;.&lt;/p&gt;'><sup>2</sup></a>.</p>
<p>Here is an example with marker MGI:1915609, where we unnest Ensembl
transcript identifiers so that we have one per row:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assoc_to_seq_ids</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">marker_id</span> <span class="op">==</span> <span class="st">"MGI:1915609"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"marker_id"</span>, <span class="st">"marker_symbol"</span>, <span class="st">"ensembl_trp_id"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unnest.html" class="external-link">unnest</a></span><span class="op">(</span><span class="st">"ensembl_trp_id"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span>n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 14 × 3</span></span></span>
<span><span class="co">##    marker_id   marker_symbol ensembl_trp_id    </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> MGI:1915609 0610010K14Rik ENSMUST00000021180</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> MGI:1915609 0610010K14Rik ENSMUST00000021181</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> MGI:1915609 0610010K14Rik ENSMUST00000100950</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> MGI:1915609 0610010K14Rik ENSMUST00000102569</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> MGI:1915609 0610010K14Rik ENSMUST00000108575</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MGI:1915609 0610010K14Rik ENSMUST00000108576</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> MGI:1915609 0610010K14Rik ENSMUST00000108577</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MGI:1915609 0610010K14Rik ENSMUST00000108578</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MGI:1915609 0610010K14Rik ENSMUST00000108579</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> MGI:1915609 0610010K14Rik ENSMUST00000134700</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> MGI:1915609 0610010K14Rik ENSMUST00000135390</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> MGI:1915609 0610010K14Rik ENSMUST00000138658</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">13</span> MGI:1915609 0610010K14Rik ENSMUST00000148878</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">14</span> MGI:1915609 0610010K14Rik ENSMUST00000150504</span></span></code></pre>
<p>Note, however, that now the meaning of each row changed: each row is
for a marker / Ensembl transcript combination.</p>
</div>
<div class="section level2">
<h2 id="variables">Variables<a class="anchor" aria-label="anchor" href="#variables"></a>
</h2>
<div class="section level3">
<h3 id="marker_status">
<code>marker_status</code><a class="anchor" aria-label="anchor" href="#marker_status"></a>
</h3>
<p><code>marker_status</code>: genetic marker status is a factor of two
levels: <code>'O'</code> for official, and <code>'W'</code> for
withdrawn. Official indicates a currently in-use genetic marker, whereas
withdrawn means that the symbol or name was once approved but has since
been replaced.</p>
</div>
<div class="section level3">
<h3 id="marker_type">
<code>marker_type</code><a class="anchor" aria-label="anchor" href="#marker_type"></a>
</h3>
<p><code>marker_type</code>: genetic marker type is a factor of 10
levels:</p>
<ul>
<li><p><code>"Gene"</code>: A locus in the cytoplasmic or nuclear genome
that is necessary and sufficient to express the complete complement of
functional products derived from a unit of transcription.</p></li>
<li><p><code>"GeneModel"</code>: A representation of an mRNA transcript
of a gene that contains information about features of the transcript
such as exon-intron boundaries, splice sites, UTRs, etc.. Due to
alternative splicing of mRNA transcripts, there may be more than one
gene model for any given gene.</p></li>
<li>
<p><code>"Pseudogene"</code>: A non-functional locus derived from a
functional locus either by:</p>
<ol style="list-style-type: decimal">
<li>replicative transfer, such as transposition, retrotransposition or
duplication,</li>
<li>mutation, where the non-functional locus is not considered an allele
of an existing functional locus in the mouse.</li>
</ol>
</li>
<li><p><code>"DNA Segment"</code>: A genomic feature recognized by
anonymous DNA probes: a segment of DNA not known to correspond to a
named gene that can be used as a marker in the construction of genetic
maps.</p></li>
<li><p><code>"Transgene"</code>: Any DNA sequence or combination of
sequences that has been introduced via a construct into the germ line of
the animal by random integration.</p></li>
<li><p><code>"QTL"</code>: Quantitative Trait Locus (QTL): the type of
marker described by statistical association to quantitative variation in
a particular phenotypic trait that is thought to be controlled by the
cumulative action of alleles at multiple loci.</p></li>
<li><p><code>"Cytogenetic Marker"</code>: A structure within a
chromosome that is visible by microscopic examination, possibly after
special staining methods are used.</p></li>
<li><p><code>"BAC/YAC end"</code>: BAC/YAC end refers to sequences at
the end of foreign DNA inserts in a BAC or YAC. These sequences are a
source of Sequence Tagged Sites (STSs) to determine the extent of
overlap between Bacterial Artificial Chromosomes (BACs) or Yeast
Artificial Chromosome (YACs) and to aid in the alignment of sequence
contigs.</p></li>
<li>
<p><code>"Complex/Cluster/Region"</code>: Refers to any of the
following:</p>
<ol style="list-style-type: decimal">
<li>Gene complex; a group of genes closely linked together that are
related evolutionarily or functionally. Interspersed unrelated genes
located within the group are included.</li>
<li>A segment of the mouse genome defined by comparison to an
orthologous segment in the genome of another species, or by some
specific characteristic, such as loss of heterozygosity.</li>
<li>A marker repository for information pertaining to a specific gene
family, where such information lacks precise family member
resolution.</li>
</ol>
</li>
<li><p><code>"Other Genome Feature"</code>: Refers to any feature of the
genome that is considered to have biological significance but that
cannot be classified with defined marker types. Major classes of other
genome features include Endogenous Viruses and Retrotransposons,
Integration Sites, and Repetitive Elements. An additional class of such
features includes genomic segments that function or are biologically
significant as DNA elements.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="marker_id">
<code>marker_id</code><a class="anchor" aria-label="anchor" href="#marker_id"></a>
</h3>
<p><code>marker_id</code>: MGI accession identifier. A unique
alphanumeric character string that is used to unambiguously identify a
particular record in the Mouse Genome Informatics database. The format
is <code>MGI:nnnnnn</code>, where <code>n</code> is a digit.</p>
</div>
<div class="section level3">
<h3 id="marker_symbol">
<code>marker_symbol</code><a class="anchor" aria-label="anchor" href="#marker_symbol"></a>
</h3>
<p><code>marker_symbol</code>: marker symbol is a unique abbreviation of
the marker name.</p>
</div>
<div class="section level3">
<h3 id="marker_name">
<code>marker_name</code><a class="anchor" aria-label="anchor" href="#marker_name"></a>
</h3>
<p><code>marker_name</code>: marker name is a word or phrase that
uniquely identifies the genetic marker, e.g. a gene or allele name.</p>
</div>
<div class="section level3">
<h3 id="feature_type">
<code>feature_type</code><a class="anchor" aria-label="anchor" href="#feature_type"></a>
</h3>
<p><code>feature_type</code>: an attribute of a portion of a genomic
sequence. See the dataset <code><a href="../reference/feature_types.html">?feature_types</a></code> for details.</p>
</div>
<div class="section level3">
<h3 id="chromosome">
<code>chromosome</code><a class="anchor" aria-label="anchor" href="#chromosome"></a>
</h3>
<p><code>chromosome</code>: mouse chromosome name. Possible values are
names for the autosomal, sexual or mitochondrial chromosomes.</p>
</div>
<div class="section level3">
<h3 id="start">
<code>start</code><a class="anchor" aria-label="anchor" href="#start"></a>
</h3>
<p><code>start</code>: genomic start position (one-offset).</p>
</div>
<div class="section level3">
<h3 id="end">
<code>end</code><a class="anchor" aria-label="anchor" href="#end"></a>
</h3>
<p><code>end</code>: genomic end position (one-offset).</p>
</div>
<div class="section level3">
<h3 id="strand">
<code>strand</code><a class="anchor" aria-label="anchor" href="#strand"></a>
</h3>
<p><code>strand</code>: DNA strand, ‘+’ for sense, and ‘-’ for
antisense.</p>
</div>
<div class="section level3">
<h3 id="genetic_map_pos">
<code>genetic_map_pos</code><a class="anchor" aria-label="anchor" href="#genetic_map_pos"></a>
</h3>
<p><code>genetic_map_pos</code>: genetic map position in centiMorgan
(cM): a unit of length in a genetic map. Two loci are 1 cM apart if
recombination is detected between them in 1% of meioses.</p>
</div>
<div class="section level3">
<h3 id="genbank_id">
<code>genbank_id</code><a class="anchor" aria-label="anchor" href="#genbank_id"></a>
</h3>
<p><code>genbank_id</code>: NCBI GenBank identifier(s), a
list-column.</p>
</div>
<div class="section level3">
<h3 id="refseq_trp_id">
<code>refseq_trp_id</code><a class="anchor" aria-label="anchor" href="#refseq_trp_id"></a>
</h3>
<p><code>refseq_trp_id</code>: NCBI RefSeq transcript identifier(s), a
list-column.</p>
</div>
<div class="section level3">
<h3 id="refseq_prt_id">
<code>refseq_prt_id</code><a class="anchor" aria-label="anchor" href="#refseq_prt_id"></a>
</h3>
<p><code>refseq_prt_id</code>: NCBI RefSeq protein identifier(s), a
list-column.</p>
</div>
<div class="section level3">
<h3 id="ensembl_trp_id">
<code>ensembl_trp_id</code><a class="anchor" aria-label="anchor" href="#ensembl_trp_id"></a>
</h3>
<p><code>ensembl_trp_id</code>: Ensembl transcript identifier(s), a
list-column.</p>
</div>
<div class="section level3">
<h3 id="ensembl_prt_id">
<code>ensembl_prt_id</code><a class="anchor" aria-label="anchor" href="#ensembl_prt_id"></a>
</h3>
<p><code>ensembl_prt_id</code>: Ensembl protein identifier(s), a
list-column.</p>
</div>
<div class="section level3">
<h3 id="swiss_prt_id">
<code>swiss_prt_id</code><a class="anchor" aria-label="anchor" href="#swiss_prt_id"></a>
</h3>
<p><code>swiss_prt_id</code>: UniProtKB/Swiss-Prot identifier(s), a
list-column.</p>
</div>
<div class="section level3">
<h3 id="tr_embl_prt_id">
<code>tr_embl_prt_id</code><a class="anchor" aria-label="anchor" href="#tr_embl_prt_id"></a>
</h3>
<p><code>tr_embl_prt_id</code>: UniProtKB/TrEMBL identifier(s), a
list-column.</p>
</div>
<div class="section level3">
<h3 id="unigene_id">
<code>unigene_id</code><a class="anchor" aria-label="anchor" href="#unigene_id"></a>
</h3>
<p><code>unigene_id</code>: NCBI UniGene identifier(s), a character
vector.</p>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ramiro Magno, David Shaw, Isabel Duarte, Ismail Gbadamosi, Ali Jawaid, Nencki Institute of Experimental Biology, University of Algarve, The Jackson Laboratory, Pattern Institute.</p>
</div>

<div class="pkgdown-footer-right">
  
</div>

    </footer>
</div>

  

  

  </body>
</html>
