<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mgi.report.reader">
<title>MGI Marker Coordinates • mgi.report.reader</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Inconsolata-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Source_Sans_Pro-0.4.9/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MGI Marker Coordinates">
<meta property="og:description" content="mgi.report.reader">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-none" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mgi.report.reader</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.9.9100</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-concepts">Concepts</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-concepts">
    <a class="dropdown-item" href="../articles/genetic_marker.html">Genetic Marker</a>
    <a class="dropdown-item" href="../articles/representative_sequence.html">Representative Genomic Sequence</a>
  </div>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-reports">Reports</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-reports">
    <h6 class="dropdown-header" data-toc-skip>Mouse Genetic Markers</h6>
    <a class="dropdown-item" href="../articles/marker_list1.html">All Markers (MRK_List1.rpt)</a>
    <a class="dropdown-item" href="../articles/marker_list2.html">Official Markers (MRK_List2.rpt)</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Sequence Data</h6>
    <a class="dropdown-item" href="../articles/marker_coordinates.html">Marker Coordinates (MGI_MRK_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_model_coordinates.html">Gene Model Coordinates (MGI_Gene_Model_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/sequence_coordinates.html">Sequence Coordinates (MGI_GTGUP.gff)</a>
    <a class="dropdown-item" href="../articles/genbank_refseq_ensembl_ids.html">GenBank, RefSeq and Ensembl IDs (MRK_Sequence.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_trembl_ids.html">UniProtKB Swiss-Prot and TrEMBL (MRK_SwissProt_TrEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_prot_ids.html">UniProtKB Swiss-Prot (MRK_SwissProt.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_trap_ids.html">Gene Traps (MRK_GeneTrap.rpt)</a>
    <a class="dropdown-item" href="../articles/ensembl_ids.html">Ensembl IDs (MRK_ENSEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/biotype_conflicts.html">Ensembl/NCBI Biotype Conflicts (MGI_BioTypeConflict.rpt)</a>
    <a class="dropdown-item" href="../articles/primers.html">Primer Pairs (PRB_PrimerSeq.rpt)</a>
    <a class="dropdown-item" href="../articles/interpro_domains.html">InterPro Domains (MGI_InterProDomains.rpt)</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/patterninstitute/mgi.report.reader/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>MGI Marker Coordinates</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/mgi.report.reader/blob/HEAD/vignettes/articles/marker_coordinates.Rmd" class="external-link"><code>vignettes/articles/marker_coordinates.Rmd</code></a></small>
      <div class="d-none name"><code>marker_coordinates.Rmd</code></div>
    </div>

    
    
<p>The MGI report <code>MGI_MRK_Coord.rpt</code> lists mouse genetic
markers along with their genomic annotations, such as chromosome
position, genome assembly version, and the source of the annotation.</p>
<p>To read this report using the key <code>"marker_coordinates"</code>,
use the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To read all records (more than 600,000), use `read_report("marker_coordinates")`.</span></span>
<span><span class="op">(</span><span class="va">marker_coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_report.html">read_report</a></span><span class="op">(</span>report_key <span class="op">=</span> <span class="st">"marker_coordinates"</span>, n_max <span class="op">=</span> <span class="fl">300L</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 300 × 12</span></span></span>
<span><span class="co">##    marker_type marker_id marker_symbol marker_name       feature_type chromosome</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Gene        MGI:87881 Acp1          acid phosphatase… protein cod… 12        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Gene        MGI:87853 a             nonagouti         protein cod… 2         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Gene        MGI:87926 Adh7          alcohol dehydrog… protein cod… 3         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Gene        MGI:87882 Acp2          acid phosphatase… protein cod… 2         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> Gene        MGI:87929 Adh5          alcohol dehydrog… protein cod… 3         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Gene        MGI:87930 Adk           adenosine kinase  protein cod… 14        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Gene        MGI:87854 Pzp           PZP, alpha-2-mac… protein cod… 6         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Gene        MGI:87859 Abl1          c-abl oncogene 1… protein cod… 2         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Gene        MGI:87883 Acp5          acid phosphatase… protein cod… 9         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Gene        MGI:87884 Acr           acrosin preprope… protein cod… 15        </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 290 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 6 more variables: start &lt;int&gt;, end &lt;int&gt;, strand &lt;fct&gt;, assembly &lt;fct&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   source &lt;fct&gt;, database &lt;fct&gt;</span></span></span></code></pre>
<p>For the first 300 records, here is the count by marker type:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">marker_coord</span>, <span class="va">marker_type</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">##   marker_type     n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Gene          286</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Pseudogene      9</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> QTL             5</span></span></code></pre>
<p>By chromosome:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">marker_coord</span>, <span class="va">chromosome</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 20 × 2</span></span></span>
<span><span class="co">##    chromosome     n</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> 1             25</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> 2             28</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> 3             26</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> 4             20</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> 5             16</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> 6             13</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> 7             20</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> 8             16</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> 9             26</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> 10            13</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> 11            15</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> 12            11</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">13</span> 13             6</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">14</span> 14            11</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">15</span> 15             8</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">16</span> 16             4</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">17</span> 17            12</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">18</span> 18             9</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">19</span> 19             9</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">20</span> X             12</span></span></code></pre>
<p>By annotation source:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">marker_coord</span>, <span class="va">source</span>, <span class="va">database</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 3</span></span></span>
<span><span class="co">##   source  database               n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> NCBI    NCBI Gene Model      158</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Ensembl Ensembl Gene Model   137</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> MGI     QTL                    5</span></span></code></pre>
<div class="section level2">
<h2 id="variables">Variables<a class="anchor" aria-label="anchor" href="#variables"></a>
</h2>
<div class="section level3">
<h3 id="marker_type">
<code>marker_type</code><a class="anchor" aria-label="anchor" href="#marker_type"></a>
</h3>
<p><code>marker_type</code>: genetic marker type is a factor of 10
levels:</p>
<ul>
<li><p><code>"Gene"</code>: A locus in the cytoplasmic or nuclear genome
that is necessary and sufficient to express the complete complement of
functional products derived from a unit of transcription.</p></li>
<li><p><code>"GeneModel"</code>: A representation of an mRNA transcript
of a gene that contains information about features of the transcript
such as exon-intron boundaries, splice sites, UTRs, etc.. Due to
alternative splicing of mRNA transcripts, there may be more than one
gene model for any given gene.</p></li>
<li>
<p><code>"Pseudogene"</code>: A non-functional locus derived from a
functional locus either by:</p>
<ol style="list-style-type: decimal">
<li>replicative transfer, such as transposition, retrotransposition or
duplication,</li>
<li>mutation, where the non-functional locus is not considered an allele
of an existing functional locus in the mouse.</li>
</ol>
</li>
<li><p><code>"DNA Segment"</code>: A genomic feature recognized by
anonymous DNA probes: a segment of DNA not known to correspond to a
named gene that can be used as a marker in the construction of genetic
maps.</p></li>
<li><p><code>"Transgene"</code>: Any DNA sequence or combination of
sequences that has been introduced via a construct into the germ line of
the animal by random integration.</p></li>
<li><p><code>"QTL"</code>: Quantitative Trait Locus (QTL): the type of
marker described by statistical association to quantitative variation in
a particular phenotypic trait that is thought to be controlled by the
cumulative action of alleles at multiple loci.</p></li>
<li><p><code>"Cytogenetic Marker"</code>: A structure within a
chromosome that is visible by microscopic examination, possibly after
special staining methods are used.</p></li>
<li><p><code>"BAC/YAC end"</code>: BAC/YAC end refers to sequences at
the end of foreign DNA inserts in a BAC or YAC. These sequences are a
source of Sequence Tagged Sites (STSs) to determine the extent of
overlap between Bacterial Artificial Chromosomes (BACs) or Yeast
Artificial Chromosome (YACs) and to aid in the alignment of sequence
contigs.</p></li>
<li>
<p><code>"Complex/Cluster/Region"</code>: Refers to any of the
following:</p>
<ol style="list-style-type: decimal">
<li>Gene complex; a group of genes closely linked together that are
related evolutionarily or functionally. Interspersed unrelated genes
located within the group are included.</li>
<li>A segment of the mouse genome defined by comparison to an
orthologous segment in the genome of another species, or by some
specific characteristic, such as loss of heterozygosity.</li>
<li>A marker repository for information pertaining to a specific gene
family, where such information lacks precise family member
resolution.</li>
</ol>
</li>
<li><p><code>"Other Genome Feature"</code>: Refers to any feature of the
genome that is considered to have biological significance but that
cannot be classified with defined marker types. Major classes of other
genome features include Endogenous Viruses and Retrotransposons,
Integration Sites, and Repetitive Elements. An additional class of such
features includes genomic segments that function or are biologically
significant as DNA elements.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="marker_id">
<code>marker_id</code><a class="anchor" aria-label="anchor" href="#marker_id"></a>
</h3>
<p><code>marker_id</code>: MGI accession identifier. A unique
alphanumeric character string that is used to unambiguously identify a
particular record in the Mouse Genome Informatics database. The format
is <code>MGI:nnnnnn</code>, where <code>n</code> is a digit.</p>
</div>
<div class="section level3">
<h3 id="marker_symbol">
<code>marker_symbol</code><a class="anchor" aria-label="anchor" href="#marker_symbol"></a>
</h3>
<p><code>marker_symbol</code>: marker symbol is a unique abbreviation of
the marker name.</p>
</div>
<div class="section level3">
<h3 id="marker_name">
<code>marker_name</code><a class="anchor" aria-label="anchor" href="#marker_name"></a>
</h3>
<p><code>marker_name</code>: marker name is a word or phrase that
uniquely identifies the genetic marker, e.g. a gene or allele name.</p>
</div>
<div class="section level3">
<h3 id="feature_type">
<code>feature_type</code><a class="anchor" aria-label="anchor" href="#feature_type"></a>
</h3>
<p><code>feature_type</code>: an attribute of a portion of a genomic
sequence. See the dataset <code><a href="../reference/feature_types.html">?feature_types</a></code> for details.</p>
</div>
<div class="section level3">
<h3 id="chromosome">
<code>chromosome</code><a class="anchor" aria-label="anchor" href="#chromosome"></a>
</h3>
<p><code>chromosome</code>: mouse chromosome name. Possible values are
names for the autosomal, sexual or mitochondrial chromosomes.</p>
</div>
<div class="section level3">
<h3 id="start">
<code>start</code><a class="anchor" aria-label="anchor" href="#start"></a>
</h3>
<p><code>start</code>: genomic start position (one-offset).</p>
</div>
<div class="section level3">
<h3 id="end">
<code>end</code><a class="anchor" aria-label="anchor" href="#end"></a>
</h3>
<p><code>end</code>: genomic end position (one-offset).</p>
</div>
<div class="section level3">
<h3 id="strand">
<code>strand</code><a class="anchor" aria-label="anchor" href="#strand"></a>
</h3>
<p><code>strand</code>: DNA strand, ‘+’ for sense, and ‘-’ for
antisense.</p>
</div>
<div class="section level3">
<h3 id="assembly">
<code>assembly</code><a class="anchor" aria-label="anchor" href="#assembly"></a>
</h3>
<p><code>assembly</code>: mouse genome assembly version, a factor of two
levels: <code>'GRCm38'</code> and <code>'GRCm39'</code>. Almost always
<code>'GRCm39'</code>.</p>
</div>
<div class="section level3">
<h3 id="source">
<code>source</code><a class="anchor" aria-label="anchor" href="#source"></a>
</h3>
<p><code>source</code>: provider of the genomic annotation.</p>
</div>
<div class="section level3">
<h3 id="database">
<code>database</code><a class="anchor" aria-label="anchor" href="#database"></a>
</h3>
<p><code>database</code>: database or catalogue within the
<code>source</code> that provides the genomic annotation.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ramiro Magno, David Shaw, Isabel Duarte, Ismail Gbadamosi, Ali Jawaid, Nencki Institute of Experimental Biology, University of Algarve, The Jackson Laboratory, Pattern Institute.</p>
</div>

<div class="pkgdown-footer-right">
  
</div>

    </footer>
</div>

  

  

  </body>
</html>
