<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mgi.report.reader">
<title>Primer Pairs • mgi.report.reader</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Inconsolata-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Source_Sans_Pro-0.4.9/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Primer Pairs">
<meta property="og:description" content="mgi.report.reader">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-none" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mgi.report.reader</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-concepts">Concepts</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-concepts">
    <a class="dropdown-item" href="../articles/genetic_marker.html">Genetic Marker</a>
    <a class="dropdown-item" href="../articles/representative_sequence.html">Representative Genomic Sequence</a>
  </div>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-reports">Reports</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-reports">
    <h6 class="dropdown-header" data-toc-skip>Mouse Genetic Markers</h6>
    <a class="dropdown-item" href="../articles/marker_list1.html">All Markers (MRK_List1.rpt)</a>
    <a class="dropdown-item" href="../articles/marker_list2.html">Official Markers (MRK_List2.rpt)</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Sequence Data</h6>
    <a class="dropdown-item" href="../articles/marker_coordinates.html">Marker Coordinates (MGI_MRK_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_model_coordinates.html">Gene Model Coordinates (MGI_Gene_Model_Coord.rpt)</a>
    <a class="dropdown-item" href="../articles/sequence_coordinates.html">Sequence Coordinates (MGI_GTGUP.gff)</a>
    <a class="dropdown-item" href="../articles/genbank_refseq_ensembl_ids.html">GenBank, RefSeq and Ensembl IDs (MRK_Sequence.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_trembl_ids.html">UniProtKB Swiss-Prot and TrEMBL (MRK_SwissProt_TrEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/swiss_prot_ids.html">UniProtKB Swiss-Prot (MRK_SwissProt.rpt)</a>
    <a class="dropdown-item" href="../articles/gene_trap_ids.html">Gene Traps (MRK_GeneTrap.rpt)</a>
    <a class="dropdown-item" href="../articles/ensembl_ids.html">Ensembl IDs (MRK_ENSEMBL.rpt)</a>
    <a class="dropdown-item" href="../articles/biotype_conflicts.html">Ensembl/NCBI Biotype Conflicts (MGI_BioTypeConflict.rpt)</a>
    <a class="dropdown-item" href="../articles/primers.html">Primer Pairs (PRB_PrimerSeq.rpt)</a>
    <a class="dropdown-item" href="../articles/interpro_domains.html">InterPro Domains (MGI_InterProDomains.rpt)</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/patterninstitute/mgi.report.reader/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Primer Pairs</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/mgi.report.reader/blob/HEAD/vignettes/articles/primers.Rmd" class="external-link"><code>vignettes/articles/primers.Rmd</code></a></small>
      <div class="d-none name"><code>primers.Rmd</code></div>
    </div>

    
    
<p>The MGI report <code>PRB_PrimerSeq.rpt</code> provides associations
between MGI genetic markers and primer pairs (probes<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;In MGI, the term &lt;em&gt;probe&lt;/em&gt; applies not only to
nucleic acid probes detected, but also to primers for PCR. This usage is
not common outside of MGI.&lt;/p&gt;"><sup>1</sup></a>).</p>
<p>To read this report using the key <code>"primers"</code>, use the
following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To read all records (more than 45,000), use `read_report("primers")`.</span></span>
<span><span class="op">(</span><span class="va">assoc_primers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_report.html">read_report</a></span><span class="op">(</span>report_key <span class="op">=</span> <span class="st">"primers"</span>, n_max <span class="op">=</span> <span class="fl">30L</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 30 × 10</span></span></span>
<span><span class="co">##    marker_id   marker_symbol marker_name    chromosome genetic_map_pos primer_id</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> MGI:1336996 10S           DNA segment, … 17                      <span style="color: #BB0000;">NA</span> MGI:1339…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> MGI:1337121 10T           DNA segment, … 17                      <span style="color: #BB0000;">NA</span> MGI:1339…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> MGI:1923536 1110006E14Rik RIKEN cDNA 11… 12                      <span style="color: #BB0000;">NA</span> MGI:5633…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> MGI:1923536 1110006E14Rik RIKEN cDNA 11… 12                      <span style="color: #BB0000;">NA</span> MGI:5633…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> MGI:1923536 1110006E14Rik RIKEN cDNA 11… 12                      <span style="color: #BB0000;">NA</span> MGI:5633…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MGI:2388983 116J6.1       DNA segment, … 7                       <span style="color: #BB0000;">NA</span> MGI:2389…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> MGI:1337061 11S           DNA segment, … 17                      <span style="color: #BB0000;">NA</span> MGI:1339…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MGI:1337057 11T           DNA segment, … 17                      <span style="color: #BB0000;">NA</span> MGI:1339…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MGI:1337123 12T           DNA segment, … 17                      <span style="color: #BB0000;">NA</span> MGI:1339…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> MGI:1337037 138017-Sp6    DNA segment, … 17                      <span style="color: #BB0000;">NA</span> MGI:1338…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 20 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 4 more variables: primer_name &lt;chr&gt;, primer_fwd_seq &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   primer_rev_seq &lt;chr&gt;, primer_amplimer_size &lt;chr&gt;</span></span></span></code></pre>
<div class="section level2">
<h2 id="primer-variables">Primer variables<a class="anchor" aria-label="anchor" href="#primer-variables"></a>
</h2>
<p>All primer related variables are scoped with the prefix
<code>primer_</code>. The <code>primer_id</code> variable is the key for
a primer pair, which is also a genetic marker, so it follows the same
format as values of <code>marker_id</code>.</p>
<div class="section level3">
<h3 id="example-mgi1923536">Example: <code>MGI:1923536</code><a class="anchor" aria-label="anchor" href="#example-mgi1923536"></a>
</h3>
<p>The primer pairs associated with marker MGI:1923536 are:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assoc_primers</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">marker_id</span> <span class="op">==</span> <span class="st">"MGI:1923536"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"marker_id"</span>, <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"primer_"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 6</span></span></span>
<span><span class="co">##   marker_id   primer_id   primer_name primer_fwd_seq      primer_rev_seq      </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> MGI:1923536 MGI:5633749 F1, R1      GTCGGTTGGAAAGGAGTGT TTGTGAAATAGCAGGCATCT</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> MGI:1923536 MGI:5633748 F3, R3      GAAAGAGGAGAAATTGGG  GTTTTCAGCACATAGTAGG </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> MGI:1923536 MGI:5633746 F2, R2      TCACATCTCAATGCCACTC CCTGTACCAGACGAAATCA </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: primer_amplimer_size &lt;chr&gt;</span></span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="variables">Variables<a class="anchor" aria-label="anchor" href="#variables"></a>
</h2>
<div class="section level3">
<h3 id="marker_id">
<code>marker_id</code><a class="anchor" aria-label="anchor" href="#marker_id"></a>
</h3>
<p><code>marker_id</code>: MGI accession identifier. A unique
alphanumeric character string that is used to unambiguously identify a
particular record in the Mouse Genome Informatics database. The format
is <code>MGI:nnnnnn</code>, where <code>n</code> is a digit.</p>
</div>
<div class="section level3">
<h3 id="marker_symbol">
<code>marker_symbol</code><a class="anchor" aria-label="anchor" href="#marker_symbol"></a>
</h3>
<p><code>marker_symbol</code>: marker symbol is a unique abbreviation of
the marker name.</p>
</div>
<div class="section level3">
<h3 id="marker_name">
<code>marker_name</code><a class="anchor" aria-label="anchor" href="#marker_name"></a>
</h3>
<p><code>marker_name</code>: marker name is a word or phrase that
uniquely identifies the genetic marker, e.g. a gene or allele name.</p>
</div>
<div class="section level3">
<h3 id="chromosome">
<code>chromosome</code><a class="anchor" aria-label="anchor" href="#chromosome"></a>
</h3>
<p><code>chromosome</code>: mouse chromosome name. Possible values are
names for the autosomal, sexual or mitochondrial chromosomes.</p>
</div>
<div class="section level3">
<h3 id="genetic_map_pos">
<code>genetic_map_pos</code><a class="anchor" aria-label="anchor" href="#genetic_map_pos"></a>
</h3>
<p><code>genetic_map_pos</code>: genetic map position in centiMorgan
(cM): a unit of length in a genetic map. Two loci are 1 cM apart if
recombination is detected between them in 1% of meioses.</p>
</div>
<div class="section level3">
<h3 id="primer_id">
<code>primer_id</code><a class="anchor" aria-label="anchor" href="#primer_id"></a>
</h3>
<p><code>primer_id</code>: an MGI accession identifier standing for a
PCR primer. A unique alphanumeric character string that is used to
unambiguously identify a particular record in the Mouse Genome
Informatics database. The format is <code>MGI:nnnnnn</code>, where
<code>n</code> is a digit.</p>
</div>
<div class="section level3">
<h3 id="primer_name">
<code>primer_name</code><a class="anchor" aria-label="anchor" href="#primer_name"></a>
</h3>
<p><code>primer_name</code>: PCR primer pair name.</p>
</div>
<div class="section level3">
<h3 id="primer_fwd_seq">
<code>primer_fwd_seq</code><a class="anchor" aria-label="anchor" href="#primer_fwd_seq"></a>
</h3>
<p><code>primer_fwd_seq</code>: PCR primer forward sequence.</p>
</div>
<div class="section level3">
<h3 id="primer_rev_seq">
<code>primer_rev_seq</code><a class="anchor" aria-label="anchor" href="#primer_rev_seq"></a>
</h3>
<p><code>primer_rev_seq</code>: PCR primer reverse sequence.</p>
</div>
<div class="section level3">
<h3 id="primer_amplimer_size">
<code>primer_amplimer_size</code><a class="anchor" aria-label="anchor" href="#primer_amplimer_size"></a>
</h3>
<p><code>primer_amplimer_size</code>: PCR primer amplimer size, as a
<strong>character vector</strong>.</p>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ramiro Magno, David Shaw, Isabel Duarte, Ismail Gbadamosi, Ali Jawaid, Nencki Institute of Experimental Biology, University of Algarve, The Jackson Laboratory, Pattern Institute.</p>
</div>

<div class="pkgdown-footer-right">
  
</div>

    </footer>
</div>

  

  

  </body>
</html>
