<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Sequence Coordinates • mgi.report.reader</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Inconsolata-0.4.9/font.css" rel="stylesheet">
<link href="../deps/Source_Sans_Pro-0.4.9/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Sequence Coordinates">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">mgi.report.reader</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-concepts" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Concepts</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-concepts">
<li><a class="dropdown-item" href="../articles/genetic_marker.html">Genetic Marker</a></li>
    <li><a class="dropdown-item" href="../articles/representative_sequence.html">Representative Genomic Sequence</a></li>
  </ul>
</li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-reports" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Reports</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-reports">
<li><h6 class="dropdown-header" data-toc-skip>Mouse Genetic Markers</h6></li>
    <li><a class="dropdown-item" href="../articles/marker_list1.html">All Markers (MRK_List1.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/marker_list2.html">Official Markers (MRK_List2.rpt)</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Sequence Data</h6></li>
    <li><a class="dropdown-item" href="../articles/marker_coordinates.html">Marker Coordinates (MGI_MRK_Coord.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/gene_model_coordinates.html">Gene Model Coordinates (MGI_Gene_Model_Coord.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/sequence_coordinates.html">Sequence Coordinates (MGI_GTGUP.gff)</a></li>
    <li><a class="dropdown-item" href="../articles/genbank_refseq_ensembl_ids.html">GenBank, RefSeq and Ensembl IDs (MRK_Sequence.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/swiss_trembl_ids.html">UniProtKB Swiss-Prot and TrEMBL (MRK_SwissProt_TrEMBL.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/swiss_prot_ids.html">UniProtKB Swiss-Prot (MRK_SwissProt.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/gene_trap_ids.html">Gene Traps (MRK_GeneTrap.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/ensembl_ids.html">Ensembl IDs (MRK_ENSEMBL.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/biotype_conflicts.html">Ensembl/NCBI Biotype Conflicts (MGI_BioTypeConflict.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/primers.html">Primer Pairs (PRB_PrimerSeq.rpt)</a></li>
    <li><a class="dropdown-item" href="../articles/interpro_domains.html">InterPro Domains (MGI_InterProDomains.rpt)</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/patterninstitute/mgi.report.reader/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Sequence Coordinates</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/mgi.report.reader/blob/main/vignettes/articles/sequence_coordinates.Rmd" class="external-link"><code>vignettes/articles/sequence_coordinates.Rmd</code></a></small>
      <div class="d-none name"><code>sequence_coordinates.Rmd</code></div>
    </div>

    
    
<p>The MGI report <code>MGI_GTGUP.gff</code><a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;GTGUP stands for Gene Trap Gene Unification Process.
This report, however, no longer includes gene trap information. At some
point the scope changed to just mean: Gene Unification Process.&lt;/p&gt;"><sup>1</sup></a> is a comprehensive
list of mouse genetic markers (&gt; 2,900,000) spanning various marker
types:</p>
<ul>
<li>Genes: ~ 55,000 markers</li>
<li>Gene Models: ~ 2,300,000 markers</li>
<li>Pseudogenes: ~ 15,000 markers</li>
<li>Transgenes: ~ 30 markers</li>
<li>Cytogenetic Markers: ~ 130 markers</li>
<li>Complex/Cluster/Region: ~ 60 markers</li>
<li>Other Genome Feature: ~ 560,000 markers</li>
</ul>
<p>To read this report using the key
<code>"sequence_coordinates"</code>, use the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To read all records (more than 2,900,000), use `read_report("sequence_coordinates")`.</span></span>
<span><span class="op">(</span><span class="va">sequence_coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_report.html">read_report</a></span><span class="op">(</span>report_key <span class="op">=</span> <span class="st">"sequence_coordinates"</span>, n_max <span class="op">=</span> <span class="fl">300L</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 300 × 9</span></span></span>
<span><span class="co">##    marker_type    marker_id marker_symbol chromosome  start    end strand source</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Other Genome … MGI:6903… Rr15647       1          3.05<span style="color: #949494;">e</span>6 3.05<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Other Genome … MGI:6914… Rr26566       1          3.05<span style="color: #949494;">e</span>6 3.05<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Other Genome … MGI:6914… Rr26567       1          3.05<span style="color: #949494;">e</span>6 3.05<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Other Genome … MGI:6903… Rr15649       1          3.06<span style="color: #949494;">e</span>6 3.06<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> Other Genome … MGI:6914… Rr26568       1          3.06<span style="color: #949494;">e</span>6 3.06<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Other Genome … MGI:6914… Rr26571       1          3.06<span style="color: #949494;">e</span>6 3.06<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Other Genome … MGI:6903… Rr15664       1          3.08<span style="color: #949494;">e</span>6 3.08<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Other Genome … MGI:6914… Rr26579       1          3.10<span style="color: #949494;">e</span>6 3.10<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Other Genome … MGI:6924… Rr36329       1          3.12<span style="color: #949494;">e</span>6 3.12<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Other Genome … MGI:6924… Rr36333       1          3.13<span style="color: #949494;">e</span>6 3.13<span style="color: #949494;">e</span>6 <span style="color: #BB0000;">NA</span>     MGI   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 290 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: feature_type &lt;fct&gt;</span></span></span></code></pre>
<div class="section level2">
<h2 id="variables">Variables<a class="anchor" aria-label="anchor" href="#variables"></a>
</h2>
<div class="section level3">
<h3 id="marker_type">
<code>marker_type</code><a class="anchor" aria-label="anchor" href="#marker_type"></a>
</h3>
<p><code>marker_type</code>: genetic marker type is a factor of 10
levels: Gene, GeneModel, Pseudogene, DNA Segment, Transgene, QTL,
Cytogenetic Marker, BAC/YAC end, Complex/Cluster/Region, Other Genome
Feature. See <code><a href="../reference/marker_type_definitions.html">?marker_type_definitions</a></code> for the meaning of
each type.</p>
</div>
<div class="section level3">
<h3 id="marker_id">
<code>marker_id</code><a class="anchor" aria-label="anchor" href="#marker_id"></a>
</h3>
<p><code>marker_id</code>: MGI accession identifier. A unique
alphanumeric character string that is used to unambiguously identify a
particular record in the Mouse Genome Informatics database. The format
is <code>MGI:nnnnnn</code>, where <code>n</code> is a digit.</p>
</div>
<div class="section level3">
<h3 id="marker_symbol">
<code>marker_symbol</code><a class="anchor" aria-label="anchor" href="#marker_symbol"></a>
</h3>
<p><code>marker_symbol</code>: marker symbol is a unique abbreviation of
the marker name.</p>
</div>
<div class="section level3">
<h3 id="marker_name">
<code>marker_name</code><a class="anchor" aria-label="anchor" href="#marker_name"></a>
</h3>
<p><code>marker_name</code>: marker name is a word or phrase that
uniquely identifies the genetic marker, e.g. a gene or allele name.</p>
</div>
<div class="section level3">
<h3 id="chromosome">
<code>chromosome</code><a class="anchor" aria-label="anchor" href="#chromosome"></a>
</h3>
<p><code>chromosome</code>: mouse chromosome name. Possible values are
names for the autosomal, sexual or mitochondrial chromosomes.</p>
</div>
<div class="section level3">
<h3 id="start">
<code>start</code><a class="anchor" aria-label="anchor" href="#start"></a>
</h3>
<p><code>start</code>: genomic start position (one-offset).</p>
</div>
<div class="section level3">
<h3 id="end">
<code>end</code><a class="anchor" aria-label="anchor" href="#end"></a>
</h3>
<p><code>end</code>: genomic end position (one-offset).</p>
</div>
<div class="section level3">
<h3 id="strand">
<code>strand</code><a class="anchor" aria-label="anchor" href="#strand"></a>
</h3>
<p><code>strand</code>: DNA strand, ‘+’ for sense, and ‘-’ for
antisense.</p>
</div>
<div class="section level3">
<h3 id="source">
<code>source</code><a class="anchor" aria-label="anchor" href="#source"></a>
</h3>
<p><code>source</code>: provider of the genomic annotation.</p>
</div>
<div class="section level3">
<h3 id="feature_type">
<code>feature_type</code><a class="anchor" aria-label="anchor" href="#feature_type"></a>
</h3>
<p><code>feature_type</code>: an attribute of a portion of a genomic
sequence. See the dataset <code><a href="../reference/feature_type_definitions.html">?feature_type_definitions</a></code> for
details.</p>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ramiro Magno, David Shaw, Isabel Duarte, Ismail Gbadamosi, Ali Jawaid, Nencki Institute of Experimental Biology, University of Algarve, The Jackson Laboratory, Pattern Institute.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer>
</div>





  </body>
</html>
